client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+import ast
+import inspect
+import re
+from importlib.metadata import packages_distributions
+from typing import Callable, List, Set, Tuple, Type, get_type_hints
+from types import ModuleType
+from verta.registry import VertaModelBase
+
+
+def list_class_functions(model_class: Type[VertaModelBase]) -> List[Tuple[str, Callable]]:
+    """List all the functions present in the provided class object."""
+    return inspect.getmembers(model_class, predicate=inspect.isfunction)
+
+
+def list_modules_in_function_definition(func: Callable) -> List[ModuleType]:
+    """List all modules called within the body of the provided function."""
+    try:
+        _func = inspect.unwrap(func)  # returns last object in chain
```

Comment:
I know we've discussed not worrying too much about wrapper for now, but just to call it out explicitly: We _might_ want to inspect the wrapper `func` itself as well for modules used, rather than jumping straight to the innermost wrapped `_func`. Could be a follow-up for next sprint.

Reply:
We could theoretically iterate through each layer with `inspect.unwrap(stop=?)`, but I agree it would be best left to future improvements.

Reply:
Captured as a potential future improvement